models/temp_table.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+var tempTableSequence uint64
+
+// createSimpleTemporaryTable creates a temporary table with the given
+// column definitions and a random name; it returns the generated name
+// and a cleanup function (which is of optional execution).
+func createSimpleTemporaryTable(sess *xorm.Session, columns string) (string, func() error, error) {
+
+	var name, create, drop string
+
+	seq := atomic.AddUint64(&tempTableSequence, 1)
```

Comment:
As we would want to enable HA (high availability), atomic ints for table names may be an issue. Not sure of 1. How big a deal this would be (as in what would possibility of conflicts), and 2. Best way to solve this (random int instead? But then you would need to calculate a random int each time, as well if you use seed for time then there is possible conflict again).

Probably not a huge issue, but figured I’d mention it just in case. (This is not a blocking review, just a fleeting thought).

Reply:
@techknowlogick The unique number id I've implemented is only for ruling out any chance of collision in case we need to use more than one table _within the same request_. Two different requests can use the same table name no problem. It's a global counter only because it's easier for me than having one implemented at the connection level.